eZ/Publish/Core/Persistence/Cache/ContentHandler.php

```diff
@@ -146,11 +146,17 @@ public function updateMetadata($contentId, MetadataUpdateStruct $struct)
     {
         $this->logger->logCall(__METHOD__, array('content' => $contentId, 'struct' => $struct));
 
-        $this->cache
-            ->getItem('content', 'info', $contentId)
-            ->set($contentInfo = $this->persistenceHandler->contentHandler()->updateMetadata($contentId, $struct))->save();
+        $contentInfo = $this->persistenceHandler->contentHandler()->updateMetadata($contentId, $struct);
 
         $this->cache->clear('content', $contentId, $contentInfo->currentVersionNo);
+        $this->cache->clear('content', 'info', $contentId);
+
+        if ($struct->remoteId) {
+            // remote id changed
+            $this->cache->clear('content', 'info', 'remoteId');
```

Comment:
In case a `remoteId` changed shouldn't we clear only the old one? (yeah, that would involve extra load)

Reply:
the cache by design does not do extra loads
this won't be an issue in 7.0 where there are cache tags

Reply:
ok, noted :)